Move baddbmm broadcasting from codegen layer to native layer.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37615 Port existing zero_dim_dispatch optimizations from codegen and remove codegen capability.
* #37614 Kill broadcasting from the codegen layer.
* #37613 Move addmm broadcasting code from codegen layer to native layer.
* #37603 Move addbmm broadcasting from the codegen layer to native layer.
* #37582 Specify _th_ ops in CUDAUnaryOps macros so they are easier to find.
* #37547 Kill the ability to codegen tensor-based broadcasting.
* #37546 Move addr broadcasting from codegen layer to native layer.
* #37545 Move broadcasting code for fmod, fmod_ from codegen layer.
* **#37544 Move baddbmm broadcasting from codegen layer to native layer.**

Differential Revision: [D21315039](https://our.internmc.facebook.com/intern/diff/D21315039)

gchanan committed:
Move baddbmm broadcasting from codegen layer to native layer.

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/native/cuda/LinearAlgebra.cu:
```diff
@@ -5,15 +5,19 @@
 
 Tensor& baddbmm_out_cuda(Tensor &result, const Tensor& self, const Tensor& batch1, const Tensor& batch2, Scalar beta, Scalar alpha) {
-  return legacy::cuda::_th_baddbmm_out(result, self, batch1, batch2, beta, alpha);
+   Tensor b_self;
```
Don't know if this is worth an update to fix, but just noting that it's here

gchanan replied:
thanks for noting, will fix at the top of the stack.

gchanan replied:
actually fixed in this PR.

gchanan committed:
Update on "Move baddbmm broadcasting from codegen layer to native layer."


Differential Revision: [D21315039](https://our.internmc.facebook.com/intern/diff/D21315039)

[ghstack-poisoned]

gchanan committed:
Update on "Move baddbmm broadcasting from codegen layer to native layer."


Differential Revision: [D21315039](https://our.internmc.facebook.com/intern/diff/D21315039)

[ghstack-poisoned]
